src/decoder.rs

```diff
@@ -107,6 +107,9 @@ impl TransactionState {
             // OUT may also be completed by NAK or STALL.
             (Some(OUT), Some(DATA0 | DATA1), NAK | STALL) => Done,
 
+            // PING may be followed by ACK or NAK.
+            (Some(PING), Some(PING), ACK | NAK) => Done,
+
```

Comment:
Should this also match on `STALL`? (based on Fig 8-28 / Fig 8-30 in the spec)

Reply:
It looks like it should, yeah.

I find figure 8-28 and the related state machine diagrams quite confusing; they seem more like they're illustrating implementation details of a reference implementation rather than the underlying protocol logic.

In the text of 8.5.1 it only says *"The endpoint either responds to the PING with a NAK or an ACK handshake"*.

But figure 8-30 is pretty clear that PING -> STALL is allowed, so I've added it.